src/core/connection.c

```diff
@@ -1850,7 +1850,9 @@ QuicConnStart(
     Status =
         QuicLibraryGetBinding(
             &UdpConfig,
-            &Path->Binding);
+            &Path->Binding,
+            Connection->State.UseQTIP,
+            Connection->State.AppDidSetQTIP);
```

Comment:
Having to have a second variable to tell if the first one should be considered or not far below is a bit unwieldy.

Could we make the value in the connection always correct, and use it as the single source of truth?
- on connection init, we can set the value based on the global ExecutionConfig
- we already prevent the app from changing the state for server connections
- lower layers simply have to do what the connection tells them to

This might maybe even let us remove the `UseTcp` setting from the datapath? (to double check...)

Reply:
I really like this!

I'll do some testing and let you know if I find anything.

Reply:
Hmm... On Connection Init the global ExecutionConfig is NULL.

Reply:
As we were discussing offline, `NULL` is a possibility if the user never set a global parameter, we can use a default.

A potentially bigger problem is synchronization... if going that route, we need to make sure we can't race with global setting change. There might already be a pattern for that, lets look around.

Reply:
Updated API to leverage QUIC_SETTINGS.